### PR TITLE
#162170690 Track comment edit history

### PR DESCRIPTION
**Pull request by luqmanoop**
#### What does this PR do?
Allows (authenticated) users edit their previous comment(s) on articles

#### Description of Task to be completed?
Have comment edit working

#### How should this be manually tested?
```bash
git clone git@github.com:andela/balder-ah-frontend.git

cd balder-ah-frontend

npm install

npm start
```
Visit an article page to comment on it and be able to edit it.

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories? (If applicable)
[#162170690](https://www.pivotaltracker.com/story/show/162170690)

#### Questions:
N/A

### CODE REVIEW

File: src/components/presentation/SmallArticleCard.jsx [hunk 1]
```diff
@@ -8,7 +8,7 @@ const SmallArticleCard = () => (
       </div>
       <div className="card-stacked">
         <div className="card-content">
-          <p>I am a very simple card. I am good at containing small bits of information.</p>
+          <p>I am a very simple card. I am good at containing small bits of information. lorem</p>
```
Comment (omokaro-faith) [inline src/components/presentation/SmallArticleCard.jsx]: Lol. Why did you have to add `lorem` here ?

File: src/actions/search.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import axios from '../utils/axiosInstance';
+import { SEARCH } from './types';
+
+
+// eslint-disable-next-line import/prefer-default-export
```
Comment (omokaro-faith) [inline src/actions/search.js]: Please remove this line.

File: src/components/comment/Comment.scss [hunk 3]
```diff
@@ -32,4 +33,23 @@
...
+        &:hover {
+            background-color: $color-light-gray;
+        }
+    }
+
+    .is-edited {
+        color: darken($color-light-gray, 30%);
+        font-size: 12px;
+    }
 }
```
Comment (omokaro-faith) [inline src/components/comment/Comment.scss]: A new line here, please.

File: src/components/comment/CommentList.test.jsx [hunk 4]
```diff
@@ -16,6 +16,26 @@ describe('<CommentList />', () => {
...
+      id: 2,
+      userId: 2,
+      body: 'Second comment',
+      edited: true,
+      author: { id: 2, username: 'janedoe', image: 'avatar.png' },
+      createdAt: '2019-01-23T23:22:33.874Z',
+    },
+    ],
+  };
+
```
Comment (omokaro-faith) [inline src/components/comment/CommentList.test.jsx]: What about abstracting this into another file.

File: src/components/comment/EditCommentForm.scss [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        border: 1px solid darken($color-danger, 5%);
+        color: $color-white;
+
+        &:hover {
+            background-color: lighten($color-danger, 5%);
+            border-color: lighten($color-danger, 5%);
+            color: darken($color-white, 2%)
+        }
+    }
+}
```
Comment (omokaro-faith) [inline src/components/comment/EditCommentForm.scss]: A new line here, please.

File: src/components/presentation/NavSearch.scss [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+        &:focus {
+            border-bottom: 1px solid $color-danger;
+            box-shadow: none;
+            padding-bottom: 2px;
+            padding-right: 25px;
+            width: 150px;
+        }
+    }
+}
```
Comment (omokaro-faith) [inline src/components/presentation/NavSearch.scss]: Here also.

File: src/components/presentation/Select.scss [hunk 7]
```diff
@@ -0,0 +1,10 @@
+@import "../../assets/sass/partials/colors";
+
+select.search-select {
+    border: 1px solid $color-danger;
+    color: $color-danger;
+    display: inline-block;
+    outline: none;
+    text-align-last:center; // https://stackoverflow.com/a/34532555
+    width: auto;
+}
```
Comment (omokaro-faith) [inline src/components/presentation/Select.scss]: ........

File: src/components/presentation/SmallArticleCard.jsx (hunk 1 above)
Comment (omokaro-faith) [reply src/components/presentation/SmallArticleCard.jsx]: This has not been modified.

**Review by omokaro-faith — CHANGES_REQUESTED** [review]

**Comment by omokaro-faith** [comment]
Please resolve conflicts also

**Review by akhilome — COMMENTED** [review]
This PR still contains two commits.

**Review by akhilome — CHANGES_REQUESTED** [review]
Please fix conflicts and issue I mentioned above so that I can properly review and approve 💪

File: src/components/presentation/NavSearch.scss (hunk 6 above)
Comment (luqmanoop) [reply src/components/presentation/NavSearch.scss]: fixed

File: src/components/comment/EditCommentForm.scss (hunk 5 above)
Comment (luqmanoop) [reply src/components/comment/EditCommentForm.scss]: fixed

File: src/components/comment/Comment.scss (hunk 3 above)
Comment (luqmanoop) [reply src/components/comment/Comment.scss]: resolved

File: src/actions/search.js (hunk 2 above)
Comment (luqmanoop) [reply src/actions/search.js]: done

File: src/components/presentation/SmallArticleCard.jsx (hunk 1 above)
Comment (luqmanoop) [reply src/components/presentation/SmallArticleCard.jsx]: resolved

**Review by akhilome — APPROVED** [review]

**Review by frostyblok — APPROVED** [review]

**Review by omokaro-faith — APPROVED** [review]

**Comment by omokaro-faith** [comment]
@codeshifu Please resolve conflicts.

**Review by oluwajuwon — APPROVED** [review]
LGTM! 👍

**Comment by omokaro-faith** [comment]
@codeshifu please resolve conflicts

**Pull request merged**